pkg/trait/knative_service.go

```diff
@@ -149,6 +150,11 @@ func (t *knativeServiceTrait) SelectControllerStrategy(e *Environment) (*Control
 		return nil, nil
 	}
 
+	// Knative serving is required
+	if ok, _ := knative.IsServingInstalled(e.Client); !ok {
```

Comment:
I missed this. We must move this into the `Configure()` func instead and add the related condition and report accordingly in order to warn the user that the trait is skipped because there is no serving API installed.

Reply:
but the controller strategy selector is a totally different interface. The `Configure()` already evaluates the controller strategy to skip the trait when the strategy is not Knative. Apart from that we need to make sure that the controller strategy is only Knative when Serving is installed as the interface can be used also when the trait is disabled. This is why I put this into the controller strategy selector interface implementation part of the trait